sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,443 @@
+   */
+  compatibilityLevel?: number;
+
+  /**
+   * Boolean value indicating whether the SQL filter expression requires preprocessing
+   */
+  requiresPreprocessing?: boolean;
+}
+
+/**
+ * @ignore RuleResourceSerializer for serializing / deserializing Rule entities
+ */
+export class RuleResourceSerializer implements AtomXmlSerializer {
+  serialize(rule: InternalRuleOptions): object {
+    const resource: { Name: any; Filter: any; Action: any } = {
+      Filter: {},
+      Action: {},
+      Name: rule.name
+    };
+
+    if (rule == undefined || rule.filter == undefined) {
+      // Defaults to creating a true filter if none specified
+      resource.Filter = {
+        SqlExpression: "1=1",
+        CompatibilityLevel: 20
+      };
+      resource.Filter[Constants.XML_METADATA_MARKER] = {
+        "p4:type": "SqlFilter",
+        "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+      };
+    } else {
+      let isSqlFilter: boolean = false;
+
+      const givenFilter: any = rule.filter as any;
+
+      if (givenFilter.sqlExpression != undefined) {
+        isSqlFilter = true;
+      } else if (
+        !givenFilter.correlationId ||
+        !givenFilter.label ||
+        !givenFilter.to ||
+        !givenFilter.replyTo ||
+        !givenFilter.replyToSessionId ||
+        !givenFilter.contentType ||
+        !givenFilter.sessionId ||
+        !givenFilter.messageId ||
+        !givenFilter.userProperties
+      ) {
+        isSqlFilter = false;
+      }
+
+      if (isSqlFilter) {
+        const sqlFilter: SqlFilter = rule.filter as SqlFilter;
+        resource.Filter = {
+          SqlExpression: sqlFilter.sqlExpression,
+          Parameters: getRawSqlParameters(sqlFilter.sqlParameters),
+          CompatibilityLevel: 20,
+          RequiresPreprocessing: getStringOrUndefined(sqlFilter.requiresPreprocessing)
+        };
+        resource.Filter[Constants.XML_METADATA_MARKER] = {
+          "p4:type": "SqlFilter",
+          "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+        };
+      } else {
+        const correlationFilter: CorrelationFilter = rule.filter as CorrelationFilter;
+
+        resource.Filter = {
+          CorrelationId: correlationFilter.correlationId,
+          Label: correlationFilter.label,
+          To: correlationFilter.to,
+          ReplyTo: correlationFilter.replyTo,
+          ReplyToSessionId: correlationFilter.replyToSessionId,
+          ContentType: correlationFilter.contentType,
+          SessionId: correlationFilter.sessionId,
+          MessageId: correlationFilter.messageId,
+          Properties: correlationFilter.userProperties
+        };
+        resource.Filter[Constants.XML_METADATA_MARKER] = {
+          "p4:type": "CorrelationFilter",
+          "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+        };
+      }
+    }
+
+    if (rule == undefined || rule.action == undefined) {
```

Comment:
The `rule == undefined` check is not needed here either for the same reason as https://github.com/Azure/azure-sdk-for-js/pull/4922#discussion_r340334672

Reply:
I would replace the above check with `if (!rule.action || !rule.action.sqlExpression)` as we need the sql expression on the action to be able to set the action

Reply:
Updated - I would keep the explicit undefined check, so that if users (JS) enter any unintended truthy value the checks wouldn't fail.